### PR TITLE
new package: incus

### PR DESCRIPTION
**Pull request by dkwo**
- I tested the changes in this PR: yes
- This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): yes

todo:
-  void docs: mention incus groups and something like `echo "root:1000000:1000000000" | doas tee -a /etc/subuid /etc/subgid`

### CODE REVIEW

**Comment by ahesford** [comment]
I don't think we should drop `lxd` in this PR.

**Comment by dkwo** [comment]
when building incus, i need to update raft, which makes it harder to bump dqlite and lxd.
the current lxd is also outdated, and it hasn't been rebuilt with at least go 1.21 as its distfiles have been moved.

**Comment by sbromberger** [comment]
I managed to get incus running on void a couple days ago. here's my runit service if it helps:

```
#!/bin/sh
exec 2>&1
_systemd_cgrp="/sys/fs/cgroup/systemd"
if [ ! -d ${_systemd_cgrp} ]; then
        mkdir ${_systemd_cgrp}
fi
if ! mountpoint -q "${_systemd_cgrp}"; then
        mount -t cgroup -o none,name=systemd cgroup ${_systemd_cgrp}
fi

[ -r conf ] && . ./conf
exec /usr/local/bin/incusd --group incus-admin --syslog ${OPTS:- --verbose}
```

**Comment by dkwo** [comment]
Thanks! Updated incus to 0.4, tried to follow upstream recommendations https://linuxcontainers.org/incus/docs/main/packaging/ which also suggests

> The incusd and incus-user daemons should be kept outside of the user’s PATH. The same is true of incus-agent which needs to be available in the daemon’s PATH but not be visible to users.

how can i keep those daemons outside of path?

**Comment by sbromberger** [comment]
> how can i keep those daemons outside of path?

Recommendation from folks on #lxc liberachat is to put them in `/usr/sbin`.

**Comment by sbromberger** [comment]
Would it be possible to make cowsql a separate PR and get that integrated sooner?

**Comment by ahesford** [comment]
Void is usr-merged and doesn't distinguish between root-visible and user-visible executables. If the executables are intended only for execution by other programs, a subdirectory of `/usr/libexec` is the proper place. Otherwise, just put them in `/usr/bin`.

**Comment by ahesford** [comment]
> is it ok to add `export PATH=/usr/libexec/incus:$PATH` to the services, so that they know where to find incus-agent?

No. If the programs that use these utilities expect them to be in the system path, then they belong in the standard `/usr/bin`. Programs that use non-user-visible executables should expect to find these utilities in `/usr/libexec` without modifications to the search path. If the upstream suggestion is to put them in `/usr/sbin` to keep them out of the default path of ordinary users (`/usr/sbin` should always be in the default path of the super user), then the answer on Void is that we make no distinction.

File: srcpkgs/incus/template [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+makedepends="lxc-devel acl-devel cowsql-devel raft-devel
+ libcap-devel libuv-devel sqlite-devel eudev-libudev-devel"
+depends="lxc acl acl-progs rsync squashfs-tools xz dnsmasq iptables attr-progs"
+short_desc="Powerful system container and virtual machine manager"
+maintainer="dkwo <npiazza@disroot.org>"
+license="Apache-2.0"
+homepage="https://linuxcontainers.org/incus"
+distfiles="https://github.com/lxc/incus/archive/refs/tags/v${version}.tar.gz"
+checksum=1195d8aecaf838806b88580dfc8b48302dcbb4612addcb6d6bfa480f14d97a0d
+system_groups="incus-admin incus"
```
Comment (leahneukirchen) [inline srcpkgs/incus/template:25]: Can we use _incus-admin and _incus here?

File: srcpkgs/incus/template [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+revision=1
+build_style=go
+go_import_path=github.com/lxc/incus
+go_build_tags=libsqlite3
+go_package="${go_import_path}/cmd/incus
+ ${go_import_path}/cmd/lxc-to-incus
+ ${go_import_path}/cmd/incus-benchmark
+ ${go_import_path}/cmd/incusd
+ ${go_import_path}/cmd/incus-user"
+# temporarily don't build fluidshift, as it conflicts with lxd, lxd-lts
```
Comment (classabbyamp) [inline srcpkgs/incus/template:13]: why not rename it to fluidshift-incus or something?

**Review by leahneukirchen — CHANGES_REQUESTED** [review]

File: srcpkgs/incus/template (hunk 1 above)
Comment (classabbyamp) [inline srcpkgs/incus/template:25]: why isn't this following the [established convention](https://github.com/void-linux/void-packages/blob/master/Manual.md#creating-system-accountsgroups-at-runtime) for system groups?

File: srcpkgs/incus/template [hunk 3]
```diff
@@ -0,0 +1,68 @@
+# Template file for 'incus'
+pkgname=incus
+version=0.4.0
+revision=1
+build_style=go
+go_import_path=github.com/lxc/incus
+go_build_tags=libsqlite3
```
Comment (classabbyamp) [inline srcpkgs/incus/template:7]: quote this

File: srcpkgs/incus/files/incus-user/check [hunk 4]
```diff
@@ -0,0 +1,2 @@
+#!/bin/sh
+incus config show >/dev/null 2>&1
```
Comment (classabbyamp) [inline srcpkgs/incus/files/incus-user/check:2]: should this be exec'd?

**Comment by classabbyamp** [comment]
squash the two incus commits please

**Review by leahneukirchen — APPROVED** [review]

**Comment by sbromberger** [comment]
Now that this has been approved, what's the process for getting it merged? (Eagerly awaiting not having to have a custom install of incus!)

**Comment by sbromberger** [comment]
Erm. Why is `lxc` a dependency here? Incus doesn't require it.

**Comment by sbromberger** [comment]
I think wget may be a transitive dependency (possibly from lxc). I installed incus from scratch on a test void system with only cowsql and raft as dependencies.

I can try my hand at a PR but I'm unfamiliar with the config files for packaging so I might mess it up. Thanks.

**Pull request merged**